starfish/imagestack/imagestack.py

```diff
@@ -50,6 +49,7 @@
 from starfish.imagestack.parser.numpy import NumpyData
 from starfish.imagestack.parser.tileset import parse_tileset
 from starfish.intensity_table.intensity_table import IntensityTable
+from starfish.multiprocessing.pool import Spool
```

Comment:
```starfish/multiprocessing/pool.py:5: error: Name 'Pool' already defined (possibly by an import)
starfish/multiprocessing/pool.py:5: error: Cycle in inheritance hierarchy```

We'd be breaking some inheritance logic, but we could change to the delegation strategy Tony mentioned if you wanted Pool

Reply:
Interesting. Thanks for exploring this. I suggest digging into why @ttung suggested that to start. It might be that the right pattern is actually governed by how we import these things, and we should name the class or function properly. The thing I _really_ care about is that the API is the same as the `stdlib` `multiprocessing.pool.Pool`. 

```python
# file: starfish/multiprocessing/pool.py

class OptionalPool:
    """Multiprocessing pool that executes on the main process when n_processes=1"""
    # code ...
```

then

```python
# file: starfish/image/filter/some_filter.py

from starfish.multiprocessing.pool import OptionalPool as Pool
# code ...
```